admin/preprints/views.py

```diff
@@ -62,6 +62,21 @@ class PreprintView(PreprintMixin, GuidView):
     """
     template_name = 'preprints/preprint.html'
     permission_required = ('osf.view_preprint', 'osf.change_preprint',)
+
+    def get_context_data(self, **kwargs):
+        preprint = self.get_object()
+        return super().get_context_data(**{
+            'preprint': preprint,
+            'SPAM_STATUS': SpamStatus,
+            'change_provider_form': ChangeProviderForm(instance=preprint),
+            'change_machine_state_form': MachineStateForm(instance=preprint),
+        }, **kwargs)
+
+
+class PreprintProviderChangeView(PreprintMixin, GuidView):
```

Comment:
Since it was easier to split the post view from the get view (`PreprintView`)  and  have it's `context_data` contain both forms and keep posting logic separate views